From 311dc53f66f669222ffff56c095e22806709ec6f Mon Sep 17 00:00:00 2001
From: Subrata Chatterjee <schatterjee@habana.ai>
Date: Thu, 10 Jun 2021 05:25:10 +0300
Subject: [PATCH] [SW-42517] greco: reset: enable fw hard reset bit

Enable CPU_BOOT_DEV_STS0_FW_HARD_RST_EN such that LKD
gets indication to invoke request to FW for hard reset.

Change-Id: I969815e9b3a119245518091bf62f410cf9fbfb14
Signed-off-by: Subrata Chatterjee <schatterjee@habana.ai>
---

diff --git a/arch/greco/common.c b/arch/greco/common.c
index 9310b79..d813bfe 100644
--- a/arch/greco/common.c
+++ b/arch/greco/common.c
@@ -818,3 +818,8 @@
 	reg |= BIT(PCIE_DBI_STATUS_COMMAND_REG_PCI_TYPE0_MEM_SPACE_EN_SHIFT);
 	regmap_writel(reg, mmPCIE_DBI_STATUS_COMMAND_REG);
 }
+
+void lkd_dev_specific_setup(void)
+{
+	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_FW_HARD_RST_EN, true);
+}
diff --git a/arch/greco_arc_gnu/common.c b/arch/greco_arc_gnu/common.c
index 602605e..836f220 100644
--- a/arch/greco_arc_gnu/common.c
+++ b/arch/greco_arc_gnu/common.c
@@ -38,6 +38,7 @@
 #include <uart.h>
 #include <string.h>
 #include <preboot/fit.h>
+#include <boot_if.h>
 #include "common.h"
 
 #include <dcon0_tstd_masks.h>
@@ -314,3 +315,8 @@
 	reg |= BIT(PCIE_DBI_STATUS_COMMAND_REG_PCI_TYPE0_MEM_SPACE_EN_SHIFT);
 	regmap_writel(reg, mmPCIE_DBI_STATUS_COMMAND_REG);
 }
+
+void lkd_dev_specific_setup(void)
+{
+	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_FW_HARD_RST_EN, true);
+}
